pkg/prompter/mock.go

```diff
@@ -0,0 +1,196 @@
+package prompter
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// PrompterMock provides stubbed out methods for prompting the user for
+// use in tests. PrompterMock has a superset of the methods on Prompter
+// so they both can satisfy the same interface.
+type PrompterMock struct {
+	t                *testing.T
+	selectStubs      []selectStub
+	multiSelectStubs []multiSelectStub
+	inputStubs       []inputStub
+	passwordStubs    []passwordStub
+	confirmStubs     []confirmStub
+}
```

Comment:
Assuming CLI extension integrators would use this in their own testing, how much documentation based on the [`cli/cli` usage](https://github.com/search?q=repo%3Acli/cli%20RegisterSelect&type=code) should be captured in documentation?

Do we expect people to depend upon https://pkg.go.dev/github.com/cli/go-gh or use markdown in the repo?

Reply:
Yeah I wanted to write up an example that compiles like the `ExamplePrompter()` but it started getting complicated due to the fact that I was trying to write a test about writing a test. Perhaps just a normal comment is enough. I will think on it a bit more.